packages/ui/src/forum/components/Thread/ThreadTitle.tsx

```diff
@@ -104,12 +104,12 @@ export const ThreadTitle = ({ thread }: ThreadTitleProps) => {
           </EditTitleInputComponent>
         </EditTitleWrapper>
       )}
-      {isMyThread && !isEditTitle && !isRemovedThread && (
+      {isMyThread && !isEditTitle && !isRemovedOrModerated && (
         <ActionButtonWrapper onClick={toggleEditTitle} size="small" square>
           <EditSymbol />
         </ActionButtonWrapper>
       )}
-      {isMyThread && !isRemovedThread && (
+      {isMyThread && !isRemovedOrModerated && (
         <ActionButtonWrapper onClick={() => deleteThread()} size="small" square>
           <DeleteIcon />
         </ActionButtonWrapper>
```

Comment:
To recap:
> it works for me as long as the thread status is not already ThreadStatusRemoved
>  the error happens only if current member != thread author (ivant != ivant-tester-9 in my case) https://github.com/Joystream/pioneer/issues/2817#issuecomment-1163133211

However we have to avoid
> The delete and edit button still show on ThreadStatusRemoved threads.
> These button also incorrectly show when the thread creator is not selected

In case the thread is moderated and the creator wants to delete the first post (and with it all posts i guess) this is not possibly because we hide the button behind the `and is not removed or moderated` flag.